tests/functional/sources/test_source_configs.py

```diff
@@ -0,0 +1,594 @@
+# expect fail - not implemented
+class TestSourceConfigsInheritence3(SourceConfigTests):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"schema.yml": all_configs_everywhere__schema_yml}
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"sources": {"tags": "project_level_important_tag"}}
+
+    @pytest.mark.xfail
+    def test_source_two_configs_source_table(self, project):
+        run_dbt(["compile"])
+        manifest = get_manifest(project.project_root)
+        assert "source.test.test_source.test_table" in manifest.sources
+        assert "source.test.test_source.other_test_table" in manifest.sources
+        config_test_table = manifest.sources.get("source.test.test_source.test_table").config
+        config_other_test_table = manifest.sources.get(
+            "source.test.test_source.other_test_table"
+        ).config
+
+        assert isinstance(config_test_table, SourceConfig)
+        assert isinstance(config_other_test_table, SourceConfig)
+
+        assert config_test_table != config_other_test_table
+        assert config_test_table == pytest.expected_config
+
+        expected_project_level_config = SourceConfig(
+            enabled=True,
+            # tags = "project_level_important_tag",   # TODO: uncomment these once SourceConfig is updated
+        )
+
+        assert config_other_test_table == expected_project_level_config
+
+
+configs_as_properties__schema_yml = """version: 2
+
+sources:
+  - name: test_source
+    quoting:
+        database: False
+        schema: False
+        identifier: False
+        column: False
+    freshness:
+        error_after: {count: 24, period: hour}
+        warn_after: {count: 12, period: hour}
+    loader: "a_loader"
+    loaded_at_field: some_column
+    database: custom_database
+    schema: custom_schema
+    # identifier: "seed"  #this doesnt seems to be supported now?
+    meta: {'languages': ['python']}
+    tags: ["important_tag"]
+    tables:
+      - name: test_table
+"""
+
+
+# Check backwards compatibility of setting configs as properties at top level
+# expect pass since the properties don't get copied to the node.config yet
+class TestSourceBackwardsCompatibility(SourceConfigTests):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"schema.yml": configs_as_properties__schema_yml}
+
+    def test_source_configs_as_properties(self, project):
+        run_dbt(["compile"])
+        manifest = get_manifest(project.project_root)
+        assert "source.test.test_source.test_table" in manifest.sources
+        config_test_table = manifest.sources.get("source.test.test_source.test_table").config
+
+        # this is new functionality - but it currently passes since SourceConfig is not updated
+        # and is commented out in the setup becuse it is not updated with new configs
+        # even when properties are defined at teh top level they should end up on the node.config
```

Comment:
> even when properties are defined at teh top level they should end up on the node.config

This should go both ways:
- When properties are defined as top-level keys (backwards compatibility for user code), they should end up in `node.config`
- Once `node.config` has been resolved, those configurations should be copied back as top-level properties (backwards compatibility for metadata consumers, including `dbt-docs` and the metadata API)

The downside is, more repetition in `metadata.json`, which is already a verbose file. I think it's worth it to ease backwards compatibility for metadata. Eventually, I could see our metadata interface including top-level node keys (well typed and validated), and excluding `node.config`, which is as `dict[Any]` as they come, and really more of an intermediate object for internal use.